src/blob.jl

```diff
@@ -59,32 +59,30 @@ Defaults to `sizeof(T)`.
     if isempty(fieldnames(T))
         quote
             $(Expr(:meta, :inline))
-            sizeof(T)
+            $(sizeof(T))
         end
     else
         quote
             $(Expr(:meta, :inline))
-            +(0, $(@splice i in 1:length(fieldnames(T)) quote
-                self_size(fieldtype(T, $i))
+            $(+(0, @splice i in 1:length(fieldnames(T)) begin
+                self_size(fieldtype(T, i))
             end))
         end
     end
 end
 
-@generated function blob_offset(::Type{T}, ::Type{Val{i}}) where {T, i}
-    quote
-        +(0, $(@splice j in 1:(i-1) quote
-            self_size(fieldtype(T, $j))
-        end))
-    end
+function blob_offset(::Type{T}, i::Int) where {T}
```

Comment:
Is the intent of this change to generate less code?  If so, how do you measure the improvement?

Reply:
This is a private function, which only gets called at code generation time of another function. Thus, it does not need to be a `@generated` function itself.